test/unit/display_utils_spec.js

```diff
@@ -35,16 +35,22 @@ describe("display_utils", function () {
       canvasFactory = null;
     });
 
-    it("`create` should throw an error if the dimensions are invalid", function () {
-      // Invalid width.
-      expect(function () {
-        return canvasFactory.create(-1, 1);
-      }).toThrow(new Error("Invalid canvas size"));
+    it("`create` should reset dimensions if they're invalid", function () {
+      if (isNodeJS) {
+        pending("Document is not supported in Node.js.");
+      }
```

Comment:
Please don't just disable testing in Node.js environments, but instead update the tests to still work there!

Reply:
Unless I'm mistaken here, is that possible here here looking at the test below it where this is also done at https://github.com/mozilla/pdf.js/pull/17908/files#diff-2b17e0587d29eb901056a2a81f40706775550a117b3d1fa7cd2a2a29e1083144R57-R59? Previously we only threw an exception which was testable in all environments, but now we're actually creating a canvas element and that requires `document` to exist. If I remove the `if` statement for the existing test, which also creates a canvas element, it fails with `TypeError: Cannot read properties of undefined (reading 'createElement'` which is probably why the existing tests are already done like this.

If there is a way in which this can be done though, that'd be great because then we can also enable the already-existing tests in Node.js I think. Unless it's easy I'm not sure it should be in scope of this PR though.

Reply:
In this case the "proper" solution, but probably best for a follow-up, would be to add a new `node_utils_spec.js` test file (only run in Node.js) with `NodeCanvasFactory` tests that are essentially a copy of these ones (modified as needed):
https://github.com/mozilla/pdf.js/blob/30b65ca0678173af8fad4cf99cc42f0d5ec2425d/test/unit/display_utils_spec.js#L27-L117